pkg/reconciler/v1alpha1/taskrun/resources/pod.go

```diff
@@ -264,3 +270,43 @@ func makeLabels(s *v1alpha1.TaskRun) map[string]string {
 	labels[pipeline.GroupName+pipeline.TaskRunLabelKey] = s.Name
 	return labels
 }
+
+// zeroNonMaxResourceRequests zeroes out the container's cpu, memory, or
+// ephemeral storage resource requests if the container does not have the
+// largest request out of all containers in the pod. This is done because Tekton
+// overwrites each container's entrypoint to make containers effectively execute
+// one at a time, so we want pods to only request the maximum resources needed
+// at any single point in time. If no contianer has an explicit resource
+// request, all requests are set to 0.
+func zeroNonMaxResourceRequests(container *corev1.Container, containerIndex int, maxIndicesByResource map[corev1.ResourceName]int) {
```

Comment:
these functions are excellent! nice focused interface and clear short functions. My only too-late-to-the-party request would be to have unit tests covering these functions directly as well but im fully expecting to be ignored since im so late haha :D

Reply:
Good point! I'm going to file an issue for a followup I thought of the other day, and if I'm in this area again I'll add some unit tests.

Reply:
awesome, that sounds great @dwnusbaum :heart: !!